### ISSUE TITLE
Currency Error after Paypal Payment

### ISSUE THREAD

**Issue by hazenetwork91** [flaviocopes/grav-plugin-shoppingcart-paypal#1]
Hi,
i'm getting the following error after every payment (using the busines sandbox account from developer.paypal.com). Everything is set to EUR.

```
RuntimeException thrown with message "Payment not successful: The transaction currency specified must be the same as previously specified."

Stacktrace:
#21 RuntimeException in /opt/lampp/htdocs/patent/user/plugins/shoppingcart-paypal/gateways/paypal_express/gateway.php:107
#20 Grav\Plugin\ShoppingCartGatewayPayPalExpress:onShoppingCartPay in /opt/lampp/htdocs/patent/user/plugins/shoppingcart-paypal/shoppingcart-paypal.php:125
#19 Grav\Plugin\ShoppingcartPaypalPlugin:onShoppingCartPay in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:181
#18 call_user_func in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:181
#17 Symfony\Component\EventDispatcher\EventDispatcher:doDispatch in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:46
#16 Symfony\Component\EventDispatcher\EventDispatcher:dispatch in /opt/lampp/htdocs/patent/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php:23
#15 RocketTheme\Toolbox\Event\EventDispatcher:dispatch in /opt/lampp/htdocs/patent/system/src/Grav/Common/Grav.php:449
#14 Grav\Common\Grav:fireEvent in /opt/lampp/htdocs/patent/user/plugins/shoppingcart-paypal/gateways/paypal_express/gateway.php:78
#13 Grav\Plugin\ShoppingCartGatewayPayPalExpress:onShoppingCartGotBackFromGateway in /opt/lampp/htdocs/patent/user/plugins/shoppingcart-paypal/shoppingcart-paypal.php:109
#12 Grav\Plugin\ShoppingcartPaypalPlugin:onShoppingCartGotBackFromGateway in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:181
#11 call_user_func in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:181
#10 Symfony\Component\EventDispatcher\EventDispatcher:doDispatch in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:46
#9 Symfony\Component\EventDispatcher\EventDispatcher:dispatch in /opt/lampp/htdocs/patent/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php:23
#8 RocketTheme\Toolbox\Event\EventDispatcher:dispatch in /opt/lampp/htdocs/patent/system/src/Grav/Common/Grav.php:449
#7 Grav\Common\Grav:fireEvent in /opt/lampp/htdocs/patent/user/plugins/shoppingcart-paypal/shoppingcart-paypal.php:65
#6 Grav\Plugin\ShoppingcartPaypalPlugin:onPluginsInitialized in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:181
#5 call_user_func in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:181
#4 Symfony\Component\EventDispatcher\EventDispatcher:doDispatch in /opt/lampp/htdocs/patent/vendor/symfony/event-dispatcher/EventDispatcher.php:46
#3 Symfony\Component\EventDispatcher\EventDispatcher:dispatch in /opt/lampp/htdocs/patent/vendor/rockettheme/toolbox/Event/src/EventDispatcher.php:23
#2 RocketTheme\Toolbox\Event\EventDispatcher:dispatch in /opt/lampp/htdocs/patent/system/src/Grav/Common/Grav.php:449
#1 Grav\Common\Grav:fireEvent in /opt/lampp/htdocs/patent/system/src/Grav/Common/Grav.php:259
#0 Grav\Common\Grav:process in /opt/lampp/htdocs/patent/index.php:38
```

in /user/plugins/shoppingcart/shoppingcart..yaml i have following config

```
`enabled: true

test_mode: true

general:
    currency: 'EUR' # The shop currency
    defaultCountry: 'AT' # Default country to be shown during Checkout
    productTaxes: 'included'

shipping:
    methods:
        -
            name: Digital shipping
            allowedCountries: ['*']
            price: 0

urls:
    checkoutURL: /checkout # The checkout page URL. Change if conflicting with another route
    saveOrderURL: /save_order # The save order page URL. Change if conflicting with another route
    orderURL: /order # The order visualization page URL. Change if conflicting with another route

ui:
    currencySymbolPosition: 'after' # Show currency `before` of `after` the price
    useOwnCSS: true # Add our own CSS. Disable if you're using your own

cart:
    allowEditingQuantityFromCart: true # Enable to allow editing a product quantity from the cart
    maximumTotalQuantityValue: # Set if you want a global limit to the ordered products
    addShippingAndTaxesCostToTotal: true #

countries: # Set countries where people can buy from, and corresponding taxes
    -
        readableName: 'Austria'
        name: 'AT'
        tax_percentage: '10' # If taxes are not factored in the product price, add them at checkout with this percentage
        allow: true
    -
        readableName: 'All unlisted countries'
        name: '*' # `*` identifies all countries
        tax_percentage: '0' # If taxes are not factored in the product price, add them at checkout with this percentage
        allow: true

checkout_form:
    name: checkout
    fields:
        -
            type: display
            content: PLUGIN_SHOPPINGCART.CHECKOUT_HEADLINE_YOUR_PERSONAL_DETAILS
            markdown: true

        -
            name: firstname
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_FIRST_NAME
            classes: js__billing__firstname
            validate:
                required: true
        -
            name: lastname
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_LAST_NAME
            classes: js__billing__lastname
            validate:
                required: true

        -
            name: email
            type: email
            label: PLUGIN_SHOPPINGCART.CHECKOUT_EMAIL
            classes: js__billing__email
            validate:
                required: true

        -
            type: display
            content: PLUGIN_SHOPPINGCART.CHECKOUT_HEADLINE_YOUR_ADDRESS
            markdown: true

        -
            name: address
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_ADDRESS
            classes: js__billing__address
            validate:
                required: true

        -
            name: telephone
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_PHONE
            classes: js__billing__telephone
            validate:
                required: true

        -
            name: city
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_CITY
            classes: js__billing__city
            validate:
                required: true

        -
            name: zip
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_ZIP
            classes: js__billing__zip
            validate:
                required: true

        -
            name: country
            type: select
            label: PLUGIN_SHOPPINGCART.CHECKOUT_COUNTRY
            classes: js__billing__country
            id: js__billing__country
            validate:
                required: true

        -
            name: state
            type: select
            label: PLUGIN_SHOPPINGCART.CHECKOUT_STATE
            classes: js__billing__state
            outerclasses: js__billing__state__control
            id: js__billing__state

        -
            name: province
            type: text
            label: PLUGIN_SHOPPINGCART.CHECKOUT_PROVINCE
            classes: js__billing__province
            outerclasses: js__billing__province__control

        -
            name: shipping_method
            type: select
            label: PLUGIN_SHOPPINGCART.CHECKOUT_CHOOSE_SHIPPING_METHOD_DESC
            classes: js__shipping__method
            outerclasses: checkout-choose-shipping-block
            id: js__shipping__method

        -
            type: display
            content: PLUGIN_SHOPPINGCART.CHECKOUT_PAYMENT_METHOD
            markdown: true
            outerclasses: checkout-choose-payment-block-title

        -
            name: payment_method
            type: select
            label: PLUGIN_SHOPPINGCART.CHECKOUT_CHOOSE_PAYMENT_METHOD_DESC
            classes: js__payment__method
            outerclasses: checkout-choose-payment-block
            id: js__payment__method

        -
            name: agree_to_terms
            type: checkbox
            ignore: true
            label: PLUGIN_SHOPPINGCART.TERMS_AND_CONDITIONS
            validate:
                required: true


    buttons:
        -
            type: submit
            value: PLUGIN_SHOPPINGCART.CHECKOUT_BUTTON_PAY
`

```
The paypal account data is set correctly in /user/plugins/shoppingcart-paypal/shoppingcart-paypal.yaml

have i overseen something?

**Comment by flaviocopes** [comment]
Mmm I have an idea, tomorrow I'll check for a possible cause of this issue.

**Comment by flaviocopes** [comment]
Recreated. I posted an update of the plugin. Could take a while to get to your site (versions are cached for a few hours), so you can install it directly from https://github.com/flaviocopes/grav-plugin-shoppingcart-paypal/zipball/1.0.2

Let me know if the issue is solved for you too.

**Comment by hazenetwork91** [comment]
Thanks, now the payment works fine!

Only one little thing.
after a sucessful transaction, when you get redirected from paypal back to the site it looks like this:
![screenshot at 2016-04-21 15 20 52](https://cloud.githubusercontent.com/assets/17365315/14710197/ad68431a-07d4-11e6-8cea-7589c403d215.png)

**Comment by flaviocopes** [comment]
Strange, maybe try clearing the grav / browser cache and retry?

**Comment by hazenetwork91** [comment]
Cleared browser and grav cache and retried it. same as before, i have also completely removed, reinstalled and reconfigured the store and paypal plugin.

![screenshot at 2016-04-21 17 18 31](https://cloud.githubusercontent.com/assets/17365315/14714550/b865a30a-07e5-11e6-9809-1a051ad0a167.png)

**Comment by flaviocopes** [comment]
Can you send me your user/ folder zipped via a Gitter PM?

**Comment by hazenetwork91** [comment]
i have sendt you a link at gitter

**Comment by flaviocopes** [comment]
Issue solved in `1.1.0-beta.1`
